### PR TITLE
chore: check if docker daemon is running for svc deploy and log the error if not

### PR DESCRIPTION
**Pull request by KollaAdithya**
**Issue**

Currently when deploying a service this is the error message when docker is not running.

>  Logging in with your password grants your terminal complete access to your account. 
> For better security, log in with a limited-privilege personal access token. Learn more at https://docs.docker.com/go/access-tokens
> Building your container image "frontyh": docker build -t 535307839156.dkr.ecr.us-west-2.amazonaws.com/ilay/frontyh:latest --platform linux/x86_64 --label com.aws.copilot.image.builder=copilot-cli --label com.aws.copilot.image.container.name=frontyh --label com.aws.copilot.image.version=v1.22.0-446-g7dc21539 /Users/kolladit/demo/web -f /Users/kolladit/demo/web/Dockerfile

     

> upload deploy resources for service frontyh: build and push the image "frontyh": build Dockerfile at /Users/kolladit/demo/web/Dockerfile: building image: exit status 1

**Fix**

With this fix we error out early before build the image. This is the exact error below if the docker daemon is not running

> Only found one service, defaulting to: frontyh
> Only found one environment, defaulting to: gray
> ✘ upload deploy resources for service frontyh: check if docker engine is running: docker daemon is not responsive: Cannot connect to the Docker daemon at unix:///Users/kolladit/.docker/run/docker.sock. Is the docker daemon running?

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

**Review by iamhopaul123 — COMMENTED** [review]
LGTM ty for the fix. we could change it from fix to chore if it wasn't an issue for our last official release.

**Review by iamhopaul123 — COMMENTED** [review]
LGTM ty for the fix. we could change it from fix to chore if it wasn't an issue for our last official release.

**Review by Lou1415926 — APPROVED** [review]
noice!

File: internal/pkg/cli/deploy/workload_test.go [hunk 1]
```diff
@@ -204,6 +205,21 @@ func TestWorkloadDeployer_UploadArtifacts(t *testing.T) {
...
+		"error if docker engine is not running": {
+			inMockUserTag: "v1.0",
+			inDockerBuildArgs: map[string]*manifest.DockerBuildArgs{
+				"mockWkld": {
+					Dockerfile: aws.String("mockDockerfile"),
+					Context:    aws.String("mockContext"),
+				},
+			},
+			mock: func(t *testing.T, m *deployMocks) {
+				m.mockdockerEngineRunChecker.EXPECT().CheckDockerEngineRunning().Return(errors.New(`docker daemon is not responsive:
```
Comment (dannyrandall) [inline internal/pkg/cli/deploy/workload_test.go]: nit: for the test case, we're just making sure that calling `UploadArtifacts()` returns an error when `CheckDockerEngineRunning` returns an error. it doesn't have to be a "real" error 😊 
```suggestion
				m.mockdockerEngineRunChecker.EXPECT().CheckDockerEngineRunning().Return(errors.New("an error")
```

**Review by dannyrandall — APPROVED** [review]
nice!

**Pull request merged**